miv/io/data.py

```diff
@@ -185,6 +188,153 @@ def set_channel_mask(self, channel_id: Iterable[int]):
+            A high value reduces type I error; a low value reduces type II error.
+            As long as this value is within a reasonable range, it should negligibly affect
+            the result (see jupyter notebook demo).
+        """
+
+        exp_binned = self.get_binned_matrix(filter, detector, offset, bins_per_second)
+        num_channels = np.shape(exp_binned[0])[1]
+
+        # if experiment is longer than spontaneous recording, it gets trunkated
+        if exp_binned[1] > spontaneous_binned[1]:
+            spontaneous_binned_copy = spontaneous_binned
+            exp_binned[0] = exp_binned[0][: spontaneous_binned[1] + 1]
+
+        # if spontaneous is longer than experiment recording
+        elif exp_binned[1] < spontaneous_binned[1]:
+            spontaneous_binned_copy = spontaneous_binned.copy()
+            spontaneous_binned_copy[0] = spontaneous_binned_copy[0][: exp_binned[1] + 1]
+
+        exp_binned_channel_rows = np.transpose(exp_binned[0])
+        spontaneous_binned_channel_rows = np.transpose(spontaneous_binned_copy[0])
+
+        dot_products = []
+        for chan in range(num_channels):
+            dot_products.append(
+                np.dot(
+                    spontaneous_binned_channel_rows[chan], exp_binned_channel_rows[chan]
+                )
+            )
+
+        mean = np.mean(dot_products)
+        threshold = mean + statistics.stdev(dot_products)
+
+        mask_list = []
+        for chan in range(num_channels):
+            if dot_products[chan] > threshold:
+                mask_list.append(chan)
+        self.add_channel_mask(np.concatenate((mask_list, exp_binned[2])))
+
+    def get_binned_matrix(
+        self,
+        filter: FilterProtocol,
+        detector: SpikeDetectionProtocol,
+        offset: float = 0,
+        bins_per_second: float = 100,
+    ):
+        """
+        Performs spike detection and return a binned 2D matrix with columns being the
+        binned number of spikes from each channel.
+
+        Parameters
+        ----------
+        filter : FilterProtocol
+            Filter that is applied to the signal before masking.
+        detector : SpikeDetectionProtocol
+            Spike detector that extracts spikes from the signals.
+        offset : float
+            The time in seconds to be trimmed in front (default = 0).
+        bins_per_second : float
+            Optional parameter for binning spikes with respect to time.
+            The spikes are binned for comparison between the spontaneous recording and
+            the other experiments. This value should be adjusted based on the firing rate.
+            A high value reduces type I error; a low value reduces type II error.
+            As long as this value is within a reasonable range, it should negligibly affect
+            the result (see jupyter notebook demo).
+
+        Returns
+        -------
+        [binned_matrix, num_bins, empty_channels]
```

Comment:
```suggestion
        binned_matrix : 
            <description>
        num_bins : int
            <description>
        empty_channels : List[int]
            <description>
```

The Sphinx (documentation tool) cannot parse this format. Please change as above.

Reply:
944a687